Lib/test/test_tabnanny.py

```diff
@@ -0,0 +1,348 @@
+    """Testing `tabnanny.errprint()`."""
+
+    def test_errprint(self):
+        """Asserting result of `tabnanny.errprint()` by giving sample inputs."""
+        tests = [
+            {'args': ['first', 'second'], 'expected': 'first second\n'},
+            {'args': ['first']          , 'expected': 'first\n'},
+            {'args': [1, 2, 3]          , 'expected': '1 2 3\n'},
+            {'args': []                 , 'expected': '\n'}
+        ]
+
+        for test in tests:
+            with captured_stderr() as stderr:
+                tabnanny.errprint(*test['args'])
+            self.assertEqual(stderr.getvalue() , test['expected'])
+
+
+class TestNannyNag(TestCase):
+    """Testing `tabnanny.NannyNag` exception."""
+    def setUp(self):
+        self.tests = [
+            {
+                'nanny': tabnanny.NannyNag(0, "foo", "bar"),
+                'expected': {
+                    'lineno': 0,
+                    'msg': 'foo',
+                    'line': 'bar'
+                }
+            },
+            {
+                'nanny': tabnanny.NannyNag(5, "testmsg", "testline"),
+                'expected': {
+                    'lineno': 5,
+                    'msg': 'testmsg',
+                    'line': 'testline'
+                }
+            }
+        ]
+
+    def test_all_methods(self):
+        """Asserting behaviour of `tabnanny.NannyNag` exception."""
+        for test in self.tests:
+            nanny = test['nanny']
+            self.assertEqual(test['expected']['lineno'], nanny.get_lineno())
+            self.assertEqual(test['expected']['msg'], nanny.get_msg())
+            self.assertEqual(test['expected']['line'], nanny.get_line())
+
+
+class TestCheck(TestCase):
+    """Testing tabnanny.check()."""
+
+    def setUp(self):
+        tabnanny.verbose = 0  # Forcefully deactivating verbose mode.
+
+    def _verify_tabnanny_check(self, dir_or_file, out="", err=""):
+        """Common verification for tabnanny.check().
+
+        Use this method to assert expected values of `stdout` and `stderr` after
+        running tabnanny.check() on given `dir` or `file` path. Because
+        tabnanny.check() captures exceptions and writes to `stdout` and
+        `stderr`, asserting standard outputs is the only way.
+
+        Arguments:
+            * dir_or_file : A path to file or directory. This argument will be
+                            fed directly to tabnanny.check().
+            * out         : Expected `stdout` after running tabnanny.check().
+            * err         : Expected `stderr` after running tabnanny.check().
+        """
+        with captured_stdout() as stdout, captured_stderr() as stderr:
+            tabnanny.check(dir_or_file)
+        self.assertEqual(stdout.getvalue(), out)
+        self.assertEqual(stderr.getvalue(), err)
+
+    def test_correct_file(self):
+        """A python source code file without any errors."""
+        file_path = create_tmp_python_file(SOURCE_CODES["error_free"])
```

Comment:
Adding it to a method will be a poor idea. If I will add it to base class then we will have a base class with one method. Will you accept that?